CMakeLists.txt

```diff
@@ -423,6 +439,15 @@ foreach(target devilution ${BIN_TARGET})
       -lfreetype -lvorbisfile -lvorbis -logg -lmodplug -lmpg123 -lSDL2 -lopusfile -lopus -lEGL -lglapi -ldrm_nouveau -lpng -lbz2 -lz -lnx)
   endif()
 
+  if(AMIGA)

```

Comment:
This could probably be something like "if(STATIC_LIBS)" and below "if (AMIGA AND NOT WARPOS)"
Shouldn't all systems statically linking SDL_ttf will need to link freetype as well?

Reply:
Why selected code was removed? It breaks 8bit on SDL1.

```
inline int SDLC_SetSurfaceAndPaletteColors(SDL_Surface *surface, SDL_Palette *palette, SDL_Color *colors, int firstcolor, int ncolors)
{
#ifdef USE_SDL1
	if (ncolors > (palette->ncolors - firstcolor)) {
		SDL_SetError("ncolors > (palette->ncolors - firstcolor)");
		return -1;
	}
	if (colors != (palette->colors + firstcolor))
		SDL_memcpy(palette->colors + firstcolor, colors, ncolors * sizeof(*colors));
	// In SDL1, the surface always has its own distinct palette, so we need to
	// update it as well.
	**#if SDL1_VIDEO_MODE_BPP == 8
		// In SDL1, if the Video Surface is 8bit, you need to set this as well to match,
		// otherwise colors will be wrong!
		SDL_SetColors(SDL_GetVideoSurface(), colors, firstcolor, ncolors);
	#endif**	
	return SDL_SetPalette(surface, SDL_LOGPAL, colors, firstcolor, ncolors) - 1;
#else
	if (SDL_SetPaletteColors(palette, colors, firstcolor, ncolors) < 0)
		return -1;
	if (surface->format->palette != palette)
		return SDL_SetSurfacePalette(surface, palette);
	return 0;
#endif
}
```

Reply:
https://www.libsdl.org/release/SDL-1.2.15/docs/html/sdlsetcolors.html
> equivalent to calling SDL_SetPalette with the flags set to (SDL_LOGPAL | SDL_PHYSPAL)

https://github.com/diasurgical/devilutionX/blob/5528cccaa604c285cb798e700d92d9618db6614c/SourceS/sdl_compat.h#L71

Reply:
In other words this should be the correct fix:
```CPP
inline int SDLC_SetSurfaceAndPaletteColors(SDL_Surface *surface, SDL_Palette *palette, SDL_Color *colors, int firstcolor, int ncolors)
{
#ifdef USE_SDL1
	if (ncolors > (palette->ncolors - firstcolor)) {
		SDL_SetError("ncolors > (palette->ncolors - firstcolor)");
		return -1;
	}
	if (colors != (palette->colors + firstcolor))
		SDL_memcpy(palette->colors + firstcolor, colors, ncolors * sizeof(*colors));
	// In SDL1, the surface always has its own distinct palette, so we need to
	// update it as well.
	int flags = SDL_LOGPAL;
#if SDL1_VIDEO_MODE_BPP == 8
	flags |= SDL_PHYSPAL;
#endif
	return SDL_SetPalette(surface, flags, colors, firstcolor, ncolors) - 1;
#else
	if (SDL_SetPaletteColors(palette, colors, firstcolor, ncolors) < 0)
		return -1;
	if (surface->format->palette != palette)
		return SDL_SetSurfacePalette(surface, palette);
	return 0;
#endif
}
```

Reply:
Or alternatly:
```
inline int SDLC_SetSurfaceAndPaletteColors(SDL_Surface *surface, SDL_Palette *palette, SDL_Color *colors, int firstcolor, int ncolors)
{
#ifdef USE_SDL1
	if (ncolors > (palette->ncolors - firstcolor)) {
		SDL_SetError("ncolors > (palette->ncolors - firstcolor)");
		return -1;
	}
	if (colors != (palette->colors + firstcolor))
		SDL_memcpy(palette->colors + firstcolor, colors, ncolors * sizeof(*colors));
#endif
	if (SDLC_SetSurfaceColors(surface, colors, firstcolor, ncolors) < 0)
		return -1;
#ifndef USE_SDL1
	if (surface->format->palette != palette)
		return SDL_SetSurfacePalette(surface, palette);
	return 0;
#endif
}
```

Reply:
Looks nicer but unfortunately doesn't work on my build.

Reply:
It's just a guess so you might have to adjust.

Reply:
@arczi84 could you submit the changes needed to fix this issue?